### ISSUE TITLE
com.docker.backend.exe is locking random files in latest stable release

### ISSUE THREAD

**Issue by Strandedpirate** [docker/for-win#5624]
- [x] I have tried with the latest version of my channel (Stable or Edge)

### Expected behavior
Files not to be randomly locked by docker.

### Actual behavior
Random files are being locked.

This script worked fine prior to install of latest stable release:

```
===============================
Removing files via git clean...
===============================
git clean -Xfd wordpress
Unlink of file 'wordpress/wp-links-opml.php' failed. Should I try again? (y/n)
```
![image](https://user-images.githubusercontent.com/1452996/73296757-f1dc3e00-41cf-11ea-92f6-8f3dac2c4a57.png)

### Information
  - Windows Version:
![image](https://user-images.githubusercontent.com/1452996/73296452-5f3b9f00-41cf-11ea-98b9-bb126b50181d.png)

  - Docker Desktop Version:
![image](https://user-images.githubusercontent.com/1452996/73296463-6498e980-41cf-11ea-925b-502ae0406329.png)

  - Are you running inside a virtualized Windows e.g. on a cloud server or on a mac VM: no

### Steps to reproduce the behavior

  1. git clean -Xfd wordpress
  2. random files are locked by com.docker.backend.exe

![image](https://user-images.githubusercontent.com/1452996/73296768-f43e9800-41cf-11ea-9d18-c9c5adc3709b.png)

**Comment by Strandedpirate** [comment]
Most likely related to issues in #5567

**Comment by spaceneenja** [comment]
also having this issue, especially using npm/yarn

**Comment by hardcoremore** [comment]
I am having the same issue with 2.2.0.3 (42716).

**Comment by mat007** [comment]
We think this has now been fixed in our master branch.
Here is a link to a Stable 2.2.0.4 RC build which fixes a few more issues related to file sharing: https://download-stage.docker.com/win/stable/43291/Docker%20Desktop%20Installer.exe
If you can reproduce your issue with it we would very much like a diagnostic (Troubleshoot windows from the systray menu) and/or reproduction steps.

Many thanks!

**Comment by hardcoremore** [comment]
File lock issue is gone but now another issue is present:

![image](https://user-images.githubusercontent.com/1597791/76403679-8c11d300-6385-11ea-8f95-20f7f33fa8c5.png)

**Comment by mat007** [comment]
@hardcoremore it’s hard to tell what this is about with just a screenshot, could you explain what you’re doing and upload a diagnostic (Troubleshot window from the systray menu)?
Thanks!

**Comment by hardcoremore** [comment]
Well I am trying just to delete the file. Both host (windows) and container are telling me that file does not exists. How can I Troubleshot window from the systray menu?

**Comment by mat007** [comment]
![image](https://user-images.githubusercontent.com/2386884/76404303-8963ad80-6386-11ea-87f6-9a81ce78a8d1.png)

I can see you’re trying to delete a file, but I have no idea how that involves Docker Desktop or what you were doing with containers before you got that issue.

**Comment by hardcoremore** [comment]
I have uploaded diagnostic. Check it out here: https://github.com/docker/for-win/issues/5985

**Comment by hardcoremore** [comment]
> I can see you’re trying to delete a file, but I have no idea how that involves Docker Desktop or what you were doing with containers before you got that issue.

File was created with php inside container. I tried to delete it from host (windows) and file is visible inside explorer but could not be deleted because it says that file could not be found.

**Comment by stephen-turner** [comment]
The original issue in this ticket is fixed in 2.2.0.4. The new issue is unrelated but is now being dealt with in #5985, so I'll close this ticket.
